docs/CodeStyle.md

```diff
@@ -32,6 +32,13 @@
 ## C++
   * used only for unit testing
   * lower-case class names (same as stl/boost)
+  * the unit tests in test/gtest are written using [C++11](https://en.cppreference.com/w/cpp/11). Whenever applicable, the usage of advanced language features is allowed and preferred over legacy code. For example:
+    * Prefer references over pointers
+    * `auto` for type deduction 
```

Comment:
I would say `auto` is not a good choice for everything since it isn't always working straight forward, for example:
```
typedef std::vector< ucs::handle<uct_ep_h> > eps_vec_t;
// ...
eps_vec_t eps;
   // fill vector
for (auto ep : eps) {
    // do smth, e.g. send a message
}
```
will destroy all eps via handle destructor what was not intent

Reply:
That is true, but it is also true for many other C++11 features ("they are complicated and there are many ways to do things wrongly with them").

But in the long-term I think we should encourage their adoption, rather than keep writing "old-style" just because of the risks.
@yosefe  WDYT?

Reply:
BTW we should replace ucs::handle with shared_ptr..

Reply:
Will do (in a separate PR).
This one has all issues resolved.